core/domain/user_services.py

```diff
@@ -1118,6 +1241,11 @@ def mark_user_for_deletion(user_id):
     user_settings = get_user_settings(user_id, strict=True)
     user_settings.deleted = True
     _save_user_settings(user_settings)
+    user_auth = _get_user_auth_from_model(
+        user_models.UserAuthModel.get_by_id(user_id))
+    if user_auth is not None:
+        user_auth.deleted = True
+        _save_user_auth_details(user_auth)
```

Comment:
Maybe leave a TODO for later when we will always have both `UserSettingsModel` and `UserAuthModel` to remove the `None` check?

Reply:
Done.